pkg/cli/image/manifest/dockercredentials/credentials.go

```diff
@@ -15,29 +15,22 @@ import (
 	"github.com/openshift/oc/pkg/helpers/image/credentialprovider"
 )
 
-var (
-	emptyKeyring = &credentialprovider.BasicDockerKeyring{}
-)
-
-// NewLocal creates a new credential store that uses the default
-// local configuration to find a valid authentication for registry
-// targets.
-func NewLocal() auth.CredentialStore {
-	keyring := &credentialprovider.BasicDockerKeyring{}
-	keyring.Add(defaultClientDockerConfig())
-	return &keyringCredentialStore{
-		DockerKeyring:     keyring,
-		RefreshTokenStore: registryclient.NewRefreshTokenStore(),
+// NewCredentialStore creates a new credential store that uses the authFilePath file
+// (defaults to a docker locations) to find a valid
+// authentication for registry targets.
+func NewCredentialStore(authFilePath string) (auth.CredentialStore, error) {
+	var cfg credentialprovider.DockerConfig
+	var err error
+
+	if authFilePath != "" {
```

Comment:
I can still move this logic to the Basic method but that would mean we would have to load the config twice because we need to notify the user when invalid config is specified (on CLI).

@dmage do you still think this is a problem? From what I see `CredentialStoreFor` is called just before making the connection